### PR TITLE
Bugfix for issue275 (fix for 'syntax error generated')

### PR DESCRIPTION
**Pull request by goto40**
Analysis branch for #275.
Proposed fix: 
 * introduced a `escale_html` for the table.
 * refactored export to separate html table export from the rest (to be tested separately)
 * added test including a simple html format check

Note:
 * For Python 2.7: the `escape_html` is only implemented rudimentary
 * Python 3.0 - 3.2 is not supported (ok)

## Code review checklist

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [x] Tests have been included and/or updated
- [n/a] Docstrings have been included and/or updated, as appropriate
- [n/a] Standalone docs have been updated accordingly
- [x] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`, no need
      to update for typo fixes and such).

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by goto40** [comment]
@igordejanovic ready for review

File: tests/functional/regressions/test_issue275.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    entity_mm = metamodel_from_str(grammar)
+
+    out_file = io.StringIO()
+    # metamodel_export(entity_mm, "test.dot")
+    renderer = DotRenderer()
+    metamodel_export_tofile(entity_mm, out_file, renderer)
+
+    text = out_file.getvalue()
+    print(text)
```
Comment (igordejanovic) [inline tests/functional/regressions/test_issue275.py]: @goto40 We shouldn't be doing printing from tests

File: tests/functional/regressions/test_issue275.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    assert text.__contains__("&lt;")
+    assert text.__contains__("&gt;")
+
+    table = renderer.get_match_rules_table()
+    print("----------")
+    print(table)
+    print("----------")
```
Comment (igordejanovic) [inline tests/functional/regressions/test_issue275.py]: Here also.

File: textx/export.py [hunk 3]
```diff
@@ -103,6 +103,14 @@ def dot_escape(s):
             .replace('?', r'\?')
 
 
+def html_escape(s):
+    if sys.version < '3':
+        return s.replace("<", "&lt;").replace(">", "&gt;")
+    else:
+        from html import escape
+        return escape(s)
+
```
Comment (igordejanovic) [inline textx/export.py:113]: This is a minor and won't make much difference in this case but it is usually a good thing to do something like:
```python
if sys.version < '3':
    def html_escape(s):
        return s.replace("<", "&lt;").replace(">", "&gt;")
else:
    def html_escape(s):
        from html import escape
        return escape(s)
```

As the condition will be evaluated only once on module import, not every time the func is called.

**Review by igordejanovic — COMMENTED** [review]
@goto40 Very good. Only some minor comments.

File: tests/functional/regressions/test_issue275.py (hunk 1 above)
Comment (goto40) [reply tests/functional/regressions/test_issue275.py]: done

File: tests/functional/regressions/test_issue275.py (hunk 2 above)
Comment (goto40) [reply tests/functional/regressions/test_issue275.py]: done

File: textx/export.py (hunk 3 above)
Comment (goto40) [reply textx/export.py:113]: done

**Review by igordejanovic — APPROVED** [review]
@goto40 LGTM. Ready for squash merge.

**Pull request merged**